scripts/WinInstallFiles/LaunchJMRI.nsi

```diff
@@ -633,38 +638,104 @@ Section "Main"
   EnvJmriHomeDone:
 
   ; -- Build the ClassPath
+  ;
+  ; The CLASSPATH shall be made up as, first to last:
+  ;  a) Any path elements specified with the -cp:p prepend argument
+  ;  b) JMRI’s code itself via the jmri.jar file
+  ;  c) contents of the program lib/ directory
+  ;  d) contents of the settings:lib/ directory 
+  ;  e) Any path elements specified with the --cp:a append argument
+
   StrCpy $CLASSPATH ".;classes"
   StrCpy $0 "$JMRIHOME" ; normally 'C:\Program Files\JMRI'
   StrCpy $3 "jmri.jar" ; set to jmri.jar to skip jmri.jar
   StrCpy $4 "" ; no prefix required
   Call GetClassPath
   StrCmp $9 "" +2 0
-  StrCpy $CLASSPATH "$CLASSPATH;$9"
+    StrCpy $CLASSPATH "$CLASSPATH;$9"
   StrCpy $CLASSPATH "$CLASSPATH;jmri.jar"
   StrCpy $3 "" ; set to blank to include all .jar files
   StrCpy $4 "lib\" ; lib prefix
   StrCpy $0 "$JMRIHOME\lib" ; normally 'C:\Program Files\JMRI\lib'
   Call GetClassPath
   StrCmp $9 "" +2 0
-  StrCpy $CLASSPATH "$CLASSPATH;$9"
+    StrCpy $CLASSPATH "$CLASSPATH;$9"
   DetailPrint "ClassPath: $CLASSPATH"
 
   ; -- Now prepend and/or append when required
   DetailPrint "Check for any prepended/appended classpath entries"
   StrCmp $P_CLASSPATH "" ClassPathAppend
-  StrCpy $CLASSPATH "$P_CLASSPATH;$CLASSPATH"
-  DetailPrint "Prepended $P_CLASSPATH"
+    StrCpy $CLASSPATH "$P_CLASSPATH;$CLASSPATH"
+    DetailPrint "Prepended $P_CLASSPATH"
+
   ClassPathAppend:
-  StrCmp $CLASSPATH_A "" ClassPathDone
-  StrCpy $CLASSPATH "$CLASSPATH;$CLASSPATH_A"
-  DetailPrint "Appended $CLASSPATH_A"
 
-  ClassPathDone:
+; add something here to "append" any/all "settings:"\lib\.jar filenames, 
+; separated by a ";".
+
+  StrCpy $0 "$PROFILE\JMRI\lib"
+  DetailPrint "Check $0 for any appended classpath entries"
+
+  Call GetSettingsClassPath
+  DetailPrint "Checked GetSettingsClassPath from $0 and found $9"
+
+  StrCmp $9 "" ContinueClasspathAppend
+    StrCpy $CLASSPATH "$CLASSPATH;$9"
+    DetailPrint "  Adding $9 to Classpath."
+
+  ContinueClasspathAppend:
+
+  StrCmp $CLASSPATH_A "" ContinueClassAppend
+    StrCpy $CLASSPATH "$CLASSPATH;$CLASSPATH_A"
+    DetailPrint "Appended $CLASSPATH_A"
+
+  ContinueClassAppend:
+
   DetailPrint "Final ClassPath: $CLASSPATH"
 
   DetailPrint "MaxLen: ${NSIS_MAX_STRLEN}"
   DetailPrint `ExeString: "$JEXEPATH" $OPTIONS -Djava.class.path="$CLASSPATH" $CLASS $PARAMETERS`
 
+  ; -- Create a preferences:\lib directory for this user
+  IfFileExists "$PROFILE\JMRI\lib\*.*" +2
+    CreateDirectory "$PROFILE\JMRI\lib"
+
+  ; -- Create the directory and README.txt file
+  IfFileExists "$PROFILE\JMRI\lib\*.*" +2
+    CreateDirectory "$PROFILE\JMRI\lib"
+
+  IfFileExists "$PROFILE\JMRI\lib\README.txt" ReadMeIsCreated
+
+  ; Create a "$JMRIPREFS\lib\README.txt" file
+  Push $0
+  FileOpen $0 "$JMRIPREFS\lib\README.txt" w
+  FileWrite $0 '(Since JMRI version 5.5.8.)$\r$\n'
```

Comment:
@devel-bobm 
The version number here, 5.5.8, needs to be updated to 5.7.1. Once that's done, I'm happy to approve and merge this PR.

Reply:
Got it.  Thanks for catching it!